service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -572,6 +573,8 @@ public ResponseEntity<ApiCloneWorkspaceResult> cloneWorkspace(
     // Construct the target workspace object from the inputs
     // Policies are cloned in the flight instead of here so that they get cleaned appropriately if
     // the flight fails.
+    List<UUID> workspaceLineage = sourceWorkspace.getWorkspaceLineage();
+    workspaceLineage.add(sourceWorkspace.getWorkspaceId());
```

Comment:
This will update `sourceWorkspace.getWorkspaceLineage()` list.
I think you might want:
```
List<UUID> workspaceLineage = new ArrayList(sourceWorkspace.getWorkspaceLineage());
```
so you make a separate list for the destination workspace.

Reply:
:O 

yikes good catch. I make it return immutable now. After reading this https://engdoc.corp.google.com/eng/doc/devguide/java/practices/choosing-collections.md?cl=head&polyglot=java#immutable. I might go do a round of immutable conversion in the repo so my future self don't accidentally makes this same mistakes again.